saleor/payment/gateways/adyen/utils.py

```diff
@@ -124,19 +85,79 @@ def request_data_for_payment(
             "currency": payment_information.currency,
         },
         "reference": payment_information.payment_id,
-        "paymentMethod": payment_data.get("paymentMethod"),
+        "paymentMethod": payment_method,
         "returnUrl": return_url,
         "merchantAccount": merchant_account,
         **extra_request_params,
     }
 
+    method = payment_method.get("type", [])
     if "klarna" in method:
-        # TODO
-        append_klarna_data(payment_information, request_data)
-
+        request_data = append_klarna_data(payment_information, request_data)
     return request_data
 
 
+def append_klarna_data(payment_information: "PaymentData", payment_data: dict):
+    _type, payment_pk = graphene.Node.from_global_id(payment_information.payment_id)
```

Comment:
Wondering if we shouldn't have database IDs in `payment_information` at this point? It looks a bit strange that payment module imports and uses Graphene which is a totally different level of responsibility. But if it requires too many changes for now than maybe it's fine.